packages/util/env.ts

```diff
@@ -30,64 +30,127 @@ if (Object.prototype.hasOwnProperty.call(process.env, 'GDAL_DATA')) {
 // Save the original process.env so we can re-use it to override
 export const originalEnv = _.cloneDeep(process.env);
 
-// Read the env-defaults for this module (relative to this typescript file)
-const envDefaults = dotenv.parse(fs.readFileSync(path.resolve(__dirname, 'env-defaults')));
-
-export let envOverrides = {};
-if (process.env.NODE_ENV !== 'test') {
-  try {
-    envOverrides = dotenv.parse(fs.readFileSync('../../.env'));
-  } catch (e) {
-    logger.warn('Could not parse environment overrides from .env file');
-    logger.warn(e.message);
+/**
+ * Parse a string env variable to a boolean or number if necessary.
+ *
+ * @param stringValue - The environment variable value as a string
+ * @returns the parsed value
+ */
+function makeConfigVar(stringValue: string): number | string | boolean {
+  if (isInteger(stringValue)) {
+    return parseInt(stringValue, 10);
+  } else if (isFloat(stringValue)) {
+    return parseFloat(stringValue);
+  } else if (isBoolean(stringValue)) {
+    return parseBoolean(stringValue);
+  } else {
+    return stringValue;
+  } 
+}
+
+/**
+  Get any errors from validating the environment - leave out the env object itself
+  from the output to avoid showing secrets.
+  @param env - the HarmonyEnv instance, including constraints
+  @returns An array of `ValidationError`s
+*/
+export function getValidationErrors(env: HarmonyEnv): ValidationError[] {
+  return validateSync(env, { validationError: { target: false } });
+}
+
+/**
+ * Get a map of image to queue URL.
+ * @param env - (Record\<string, string\>) containing all environment config properties,
+ * with snake-cased keys
+ */
+function queueUrlsMap(env: Record<string, string>): Record<string, string> {
+  // process all environment variables ending in _QUEUE_URLS to add image/url pairs to
+  // the `serviceQueueUrls` map
+  const serviceQueueUrls = {};
+  for (const k of Object.keys(env)) {
+    if (/^.*_QUEUE_URLS$/.test(k)) {
+      const value = env[k];
+      try {
+        const imageQueueUrls = JSON.parse(value);
+        for (const imageQueueUrl of imageQueueUrls) {
+          const [image, url] = imageQueueUrl.split(',');
+          if (image && url) {
+            // replace 'localstack' with `env.localstackHost` to allow for harmony to be run in a
+            // container
+            serviceQueueUrls[image] = url.replace('localstack', env.LOCALSTACK_HOST);
+          }
+        }
+      } catch (e) {
+        logger.error(`Could not parse value ${value} for ${k} as JSON`);
+      }
+    }
   }
+  return serviceQueueUrls;
 }
 
-export interface IHarmonyEnv {
```

Comment:
Does dropping these definitions mean we do not have type information for the environment variables or a list of the valid variables that can be used from the `env` object? I think it's ok and we can still add validations, but just sanity checking this is desired.

Reply:
You'll still get both.